File: src/beginner/scripts/add_three_ints_client.py

import sys
import rospy
from beginner.srv import AddThreeInts, AddThreeIntsRequest

def add_three_ints_client(x, y, z):
    rospy.wait_for_service('add_three_ints')
    try:
        add_three_ints = rospy.ServiceProxy('add_three_ints', AddThreeInts)
        resp1 = add_three_ints(x, y, z)
        return resp1.ans
    except (rospy.ServiceException):
        print ("Service call failed:")

def usage():
    return ("%s [x y z]" %sys.argv[0])

if __name__ == "__main__":
    print(sys.argv)
    if(len(sys.argv) == 4):
        x = int(sys.argv[1])
        y = int(sys.argv[2])
        z = int(sys.argv[3])
    else:
        print (usage())
        sys.exit(1)
    print ("Requesting %s + %s + %s" %(x, y, z))
    print ("%s + %s + %s = %s" %(x, y, z, add_three_ints_client(x, y, z)))

File: src/beginner/scripts/add_three_ints_client.cpp
#include <ros/ros.h>
#include <beginner/AddThreeInts.h>

void add_three_ints_clients(int x, int y, int z, int &answer){

}

int main(int argc, char**argv){

    ros::init(argc,argv,"add_three_ints_client");

    if(argc !=4){
        ROS_INFO("usage : add_three_ints_client x y z");
        return 1;
    }

    ros::NodeHandle n;
    ros::ServiceClient client = n.serviceClient<beginner::AddThreeInts>("add_three_ints");
    beginner::AddThreeInts srv;
    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);
    srv.request.c = atoll(argv[3]);
    if(client.call(srv)){
        ROS_INFO("Sum: %ld", (long int)srv.response.ans);
    }
    else{
        ROS_ERROR("failed to call service add_three_ints");
    }
}